lms/djangoapps/instructor_task/tasks_helper/module_state.py

```
@@ -174,84 +175,50 @@ def rescore_problem_module_state(xmodule_instance_args, module_descriptor, stude
-                dict(
-                    course=course_id,
-                    loc=usage_key,
-                    student=student
-                )
-            )
-
-            if result is not None:  # Only for CAPA. This will get moved to the grade handler.
-                new_weighted_earned, new_weighted_possible = weighted_score(
-                    result['new_raw_earned'] if result else None,
-                    result['new_raw_possible'] if result else None,
-                    module_descriptor.weight,
-                )
-
-                # TODO: remove this context manager after completion of AN-6134
-                context = course_context_from_course_id(course_id)
-                with tracker.get_tracker().context(GRADES_RESCORE_EVENT_TYPE, context):
-                    tracker.emit(
-                        unicode(GRADES_RESCORE_EVENT_TYPE),
-                        {
-                            'course_id': unicode(course_id),
-                            'user_id': unicode(student.id),
-                            'problem_id': unicode(usage_key),
-                            'new_weighted_earned': new_weighted_earned,
-                            'new_weighted_possible': new_weighted_possible,
-                            'only_if_higher': task_input['only_if_higher'],
-                            'instructor_id': unicode(xmodule_instance_args['request_info']['user_id']),
-                            'event_transaction_id': unicode(event_transaction_id),
-                            'event_transaction_type': unicode(GRADES_RESCORE_EVENT_TYPE),
-                        }
-                    )
-            return UPDATE_STATUS_SUCCEEDED
-        else:
+        # specific events from CAPA are not propagated up the stack. Do we want this?
+        try:
+            instance.rescore(only_if_higher=task_input['only_if_higher'])
+        except (LoncapaProblemError, StudentInputError, ResponseError):
```

This section here changed because there is no longer a `result` object to inspect- the logic here is now based around exception handling. The `ScorableXBlockMixin` doesn't specify any return object for `rescore`s so I think this is how it has to be. I did consider swallowing this exception lower down but had a hard time differentiating between successful and failed rescore attempts when using that design.

TLDR: rescores that raise these three exceptions are now treated as failures instead of successes.